n3fit/src/n3fit/scripts/n3fit_exec.py

```diff
@@ -145,6 +145,12 @@ def from_yaml(cls, o, *args, **kwargs):
             validation_action = namespace + "validation_pseudodata"
 
             N3FIT_FIXED_CONFIG['actions_'].extend((training_action, validation_action))
+        if (thconfig:=file_content.get('theory')) is not None:
+            N3FIT_FIXED_CONFIG['theoryid']=thconfig.get('theoryid', True)
```

Comment:
I'm confused. What is theory id for you? If it is the theory id number we already have that information within n3fit (maybe not used explicitly).

You might want to create a vp action that generates the photon so that `n3fit` can depend in its argument of `photon_fiatlux,`

where `photon_fiatlux` depends on `(theory, fiatlux)`

And vp will automatically pass the `theory` and the `fiatlux` parts of the runcard to the action `photon_fiatlux` (so you don't need to change this). The result of `photon_fiatlux` can be `None` when `fiatlux` is not present for instance.

Reply:
Here I didn't know how to pass the ```theoryid``` to ```photon_fitting_scale``` that is called inside ```model_trainer```. Probably there is a better way